make_site.py

```diff
@@ -455,12 +457,48 @@ class User:
     fullname: str
     lon: float
     lat: float
-    zulip: str
     github: Optional[str] = None
     website: Optional[str] = None
 
-with (DATA/'users.yaml').open('r', encoding='utf-8') as h_file:
-    users = [User(**e) for e in yaml.safe_load(h_file)]
+client = zulip.Client(
+    email='map-scraper-bot@leanprover.zulipchat.com', 
+    site='https://leanprover.zulipchat.com', 
+    api_key=os.environ.get('ZULIP_KEY')) if 'ZULIP_KEY' in os.environ else None
+
+# Zulip custom profile fields are tracked by ID, not by name
+profile_data_fields = {
+    'website': '5150',
+    'latitude': '5148',
+    'longitude': '5149',
+    'github': '3658'
+}
+
+def get_user_field(user, field):
+    try:
+        return user['profile_data'][profile_data_fields[field]]['value']
+    except KeyError:
+        return None
```

Comment:
nit:
```suggestion
    field_id = profile_data_fields[field]
    try:
        return user['profile_data'][field_id]['value']
    except KeyError:
        return None
```
so that typos in field names aren't silenced.